src/main/java/ch/njol/skript/command/ScriptCommand.java

```diff
@@ -246,20 +246,16 @@ public boolean execute(final CommandSender sender, final String commandLabel, fi
 			}
 		}
 
-		if (Bukkit.isPrimaryThread()) {
-			execute2(event, sender, commandLabel, rest);
-			if (sender instanceof Player && !event.isCooldownCancelled())
+		Runnable runnable = () -> {
+			if (sender instanceof Player)
 				setLastUsage(((Player) sender).getUniqueId(), event, new Date());
+			execute2(event, sender, commandLabel, rest);
```

Comment:
Does executing after setting last usage mess with https://github.com/SkriptLang/Skript/blob/e4dc9ef722916dedb74590548b260fc8c84c09c9/src/main/java/ch/njol/skript/expressions/ExprCmdCooldownInfo.java#L62?

Reply:
Yes, it does. With the previous code, the `last usage` expression would return null in the trigger sections:
```
command /test:
	cooldown: 2 seconds
	trigger:
		broadcast "c1 %last usage%"
		wait 1 tick
		broadcast "c2 %last usage%"
```
Without changes:
![image](https://user-images.githubusercontent.com/29547183/139854635-1244b6a8-81ac-4ac6-918b-331f20f2119e.png)
With changes:
![image](https://user-images.githubusercontent.com/29547183/139855331-950fc371-a46a-41b4-97c0-afcb48fc4470.png)

I personally think the latter, with changes, makes more sense

Reply:
Does it make sense for it to just to function as the now expression essentially? How does it affect https://github.com/SkriptLang/Skript/blob/f54baf226240d7eda0ba1489b0b46676f7adaf02/src/main/resources/config.sk#L176? I just want to make sure this won't cause any unintended side effects 🙂

Reply:
I think it makes sense to set the cooldown as soon as the command is used, as there is nothing that changed in the waited 1 tick (referring to test script I sent) that would make the cooldown change. The last usage expression is mostly made for the cooldown message entry anyway.

Same code, master branch:
![image](https://user-images.githubusercontent.com/29547183/139860384-20c241ec-88ca-471f-9bbc-2c4377fa1466.png)
PR branch:
![image](https://user-images.githubusercontent.com/29547183/139861171-8bbda46c-5ffb-4fac-b883-94fb94ebfd69.png)

> I just want to make sure this won't cause any unintended side effects 🙂

No worries :D

Reply:
Yeah, I see what you mean with the delays. 

It could be useful for users though if we keep this behavior (so they don't have to track "actual" last usage times themselves)

Reply:
That's true, but that doesn't have much to do with cooldowns, and this feature is currently only for commands with cooldowns

Reply:
I just don't know if we should change the behavior here. To me, it makes more sense that `last usage` should return the last time the command was executed. I think it's okay for the delay to make a difference as time has then passed since the user executed the command.

It may even be a good idea to expand this expression past commands with cooldowns, but that part is out of the scope of this PR.

Reply:
I'll return the behaviour to its original 👍